torch_geometric/nn/conv/pna_conv.py

```diff
@@ -181,27 +179,7 @@ def message(self, x_i: Tensor, x_j: Tensor,
     def aggregate(self, inputs: Tensor, index: Tensor,
                   dim_size: Optional[int] = None) -> Tensor:
 
-        outs = []
-        for aggregator in self.aggregators:
-            if aggregator == 'sum':
-                out = scatter(inputs, index, 0, None, dim_size, reduce='sum')
-            elif aggregator == 'mean':
-                out = scatter(inputs, index, 0, None, dim_size, reduce='mean')
-            elif aggregator == 'min':
-                out = scatter(inputs, index, 0, None, dim_size, reduce='min')
-            elif aggregator == 'max':
-                out = scatter(inputs, index, 0, None, dim_size, reduce='max')
-            elif aggregator == 'var' or aggregator == 'std':
-                mean = scatter(inputs, index, 0, None, dim_size, reduce='mean')
-                mean_squares = scatter(inputs * inputs, index, 0, None,
-                                       dim_size, reduce='mean')
-                out = mean_squares - mean * mean
-                if aggregator == 'std':
-                    out = torch.sqrt(torch.relu(out) + 1e-5)
-            else:
-                raise ValueError(f'Unknown aggregator "{aggregator}".')
-            outs.append(out)
-        out = torch.cat(outs, dim=-1)
+        out = super().aggregate(inputs, index, dim_size=dim_size)
 
         deg = degree(index, dim_size, dtype=inputs.dtype)
```

Comment:
Can we move these scalers outside `aggregate`? This way, we don't have to override `aggregate` at all.

Reply:
Oh good idea :-)

Reply:
I updated to apply the scalers as a hook - WDYT?

Reply:
Interesting solution, although I personally feel we shouldn't add business logic to hooks TBH. If you are not a fan of moving this logic post `self.propagate`, we could add it as an additional module to `nn.aggr`, e.g.,
```python
class DegreeScalerAggregation(Aggregation):
    def __init__(self, aggregator, scalers, deg):
        pass
```

Reply:
Alternatively, I may be mistaken to move the logic outside `aggregate` - This should definitely be part of aggregation.

Reply:
Actually I like the degree scaler as an  Aggregation. It keeps it within the Aggregation logic and should be eaiser to reuse.

Agree on the hooks not feeling like a great solution.

Reply:
Agree with adding the degree scaler as an `Aggregation` is better. Alternatively, we can add it as a helper method to the `Aggregation` base class instead of an `Aggregation`.

Reply:
I've made changes for this now. I'm leaving some comments in the diff though where I'd like some input